client/my-sites/domains/domain-management/list/bulk-all-domains.tsx

```diff
@@ -41,13 +50,49 @@ export default function BulkAllDomains( props: BulkAllDomainsProps ) {
 		<OptionsDomainButton key="breadcrumb_button_1" specificSiteActions allDomainsList />,
 	];
 
+	const isCreditCardExpiring = ( purchases: Purchase[] ) => ( domain: ResponseDomain ) => {
```

Comment:
We will need to get the purchases to process it in the domains table. Instead of passing down purchases and then finding the purchase for each domain, I'm doing it from the composition root with the methods that use it and then pass those methods down. Once we get all purchases from here we pass it to these methods

Reply:
Yeah It's already in redux store https://github.com/Automattic/wp-calypso/blob/f5183659d16df7c8826033fc7a2eeeb259c89bfa/client/my-sites/domains/domain-management/list/all-domains.jsx#L774 

I set it to empty for now.